src/components/header/header_section/header_section_item_button.tsx

```diff
@@ -61,15 +60,147 @@ export const EuiHeaderSectionItemButton = forwardRef<
       className,
       notification,
       notificationColor = 'accent',
-      animation = false,
       ...rest
     },
     ref
   ) => {
+    const [buttonRef, setButtonRef] = useState<EuiHeaderSectionItemButtonRef>();
+    const animationTargetRef = useRef<HTMLSpanElement | null>(null);
+
+    useImperativeHandle<
+      EuiHeaderSectionItemButtonRef,
+      EuiHeaderSectionItemButtonRef
+    >(
+      ref,
+      () => {
+        if (buttonRef) {
+          (buttonRef as any).triggerAnimation = () => {
+            const keyframes: Keyframe[] = [
```

Comment:
This is converted straight from the CSS animation. We can probably find a small math solution to generating these in a loop instead. Either way is fine.